tooling/image-updater/internal/output/output.go

```diff
@@ -0,0 +1,165 @@
+
+package output
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/template"
+
+	"github.com/Azure/ARO-HCP/tooling/image-updater/internal/yaml"
+)
+
+// summaryTemplate is a Go template for the summary output.
+// Available fields:
+//   - .TotalImages: total number of images processed
+//   - .UpdatedCount: number of images updated
+//   - .DryRun: whether this is a dry-run
+const summaryTemplate = `
+Summary
+-------
+{{- if .DryRun }}
+Total images checked: {{ .TotalImages }}
+Updates available:    {{ .UpdatedCount }}
+
+This was a dry-run. No files were modified.
+{{- else }}
+Total images checked: {{ .TotalImages }}
+Images updated:       {{ .UpdatedCount }}
+{{- end }}
+`
+
+// commitMessageTemplate is a Go template for the commit message.
+// Available fields in .Updates array:
+//   - .Name: image name
+//   - .OldSHA: old digest (truncated)
+//   - .NewSHA: new digest (truncated)
+//   - .Version: version tag
+//   - .Timestamp: build timestamp
+const commitMessageTemplate = `Updated images in target config files:
+
+| Image | Old SHA | New SHA | Version | Timestamp |
+|-------|---------|---------|---------|-----------|
+{{- range .Updates }}
+| {{ .Name }} | {{ .OldSHA }} | {{ .NewSHA }} | {{ .Version }} | {{ .Timestamp }} |
+{{- end }}
+`
+
+// summaryData holds data for the summary template
+type summaryData struct {
+	TotalImages  int
+	UpdatedCount int
+	DryRun       bool
+}
+
+// updateData holds data for a single update in the commit message
+type updateData struct {
+	Name      string
+	OldSHA    string
+	NewSHA    string
+	Version   string
+	Timestamp string
+}
+
+// commitMessageData holds data for the commit message template
+type commitMessageData struct {
+	Updates []updateData
+}
+
+// PrintSummary prints a formatted summary of the update operation
+func PrintSummary(w io.Writer, totalImages, updatedCount int, dryRun bool) {
+	tmpl, err := template.New("summary").Parse(summaryTemplate)
+	if err != nil {
+		fmt.Fprintf(w, "error parsing summary template: %v\n", err)
+		return
+	}
+
+	data := summaryData{
+		TotalImages:  totalImages,
+		UpdatedCount: updatedCount,
+		DryRun:       dryRun,
+	}
+
+	if err := tmpl.Execute(w, data); err != nil {
+		fmt.Fprintf(w, "error executing summary template: %v\n", err)
+	}
+}
+
+// GenerateCommitMessage creates a markdown table commit message for the updated images
+func GenerateCommitMessage(updates map[string][]yaml.Update) string {
+	var allUpdates []yaml.Update
+	for _, updates := range updates {
+		allUpdates = append(allUpdates, updates...)
+	}
+
+	if len(allUpdates) == 0 {
+		return ""
+	}
+
+	// Deduplicate updates by image name
+	seen := make(map[string]bool)
+	var uniqueUpdates []updateData
+	for _, update := range allUpdates {
+		if !seen[update.Name] {
+			seen[update.Name] = true
+
+			// Strip sha256: prefix if present, then take first 7 chars
+			oldSHA := strings.TrimPrefix(update.OldDigest, "sha256:")
+			if len(oldSHA) > 7 {
+				oldSHA = oldSHA[:7] + "…"
+			}
```

Comment:
7 is hardcoded in multiple places. Let's use a constant instead e.g.  const shortSHALength = 7

Reply:
7 is usually the standard to short shas, but we can add a constant.